compiler+runtime/include/cpp/jank/read/parse.hpp

```diff
@@ -12,33 +14,75 @@ namespace jank::runtime
 /* TODO: Rename file to processor. */
 namespace jank::read::parse
 {
-  static option<char> get_char_from_literal(native_persistent_string const &sv)
+  static option<native_persistent_string>
+  parse_character_in_base(native_persistent_string_view const &char_literal, int const base)
+  {
+    try
+    {
+      size_t chars_processed{};
+      auto const codepoint(
+        std::stol(native_persistent_string{ char_literal.data() + 2, char_literal.size() - 2 },
+                  &chars_processed,
+                  base));
+
+      /* Some characters that weren't processed at all.
+       * An example would be when trying to parse `12a` in octal base,
+       * `stol` will only parse `12` and will ignore `a`, returning `2`
+       * in `chars_processed`.
+       * We return none in that case, to represent to the downstream callers
+       * the enability to parse the char literal.
+       *
+       * Refer: https://en.cppreference.com/w/cpp/string/basic_string/stol
+       */
+      if(chars_processed != char_literal.size() - 2)
+      {
+        return none;
+      }
+
+      std::wstring_convert<std::codecvt_utf8<char32_t>, char32_t> converter;
+      return native_persistent_string{ converter.to_bytes(codepoint) };
+    }
+    catch(std::range_error const)
+    {
+      return none;
+    }
+    catch(std::invalid_argument const)
+    {
+      return none;
+    }
+    catch(std::out_of_range const)
+    {
+      return none;
+    }
```

Comment:
Just one `catch(std::exception const &)` will do. Always catch exceptions by reference.

Reply:
`catch(std::exception const &)` will catch all exceptions. Even the ones thrown not because of wrong syntax.

Why is that better in that case?

Reply:
We ant all exceptions caught here, since we're encoding error handling in the return type. Having an `option` or a `result` as a return type AND having to deal with exceptions is the worst of both worlds, in terms of API design.